airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -76,7 +76,7 @@ def compute_slices(self) -> Iterable[Optional[Mapping[str, Any]]]:
         slices = [{"files": list(group[1])} for group in itertools.groupby(sorted_files_to_read, lambda f: f.last_modified)]
         return slices
 
-    def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[AirbyteMessage]:
+    def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[AirbyteMessage]:  # type: ignore # file based stream returns an AirbyteMessage while other implementations return mapping
```

Comment:
`read_records_from_slice` only seems to exist for file based connectors. can we align the returned so it always return either AirbyteMessage or Mapping?


```
git grep read_records_from_slice | cat
airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/abstract_file_based_stream.py:        return self.read_records_from_slice(stream_slice)
airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/abstract_file_based_stream.py:    def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[Mapping[str, Any]]:
airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py:    def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[Mapping[str, Any]]:
airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py:        yield from self._legacy_stream.read_records_from_slice(stream_slice)
airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py:    def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[AirbyteMessage]:
airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py:    def test_when_read_records_from_slice_then_return_records(self) -> None:
airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py:        messages = list(self._stream.read_records_from_slice({"files": [RemoteFile(uri="uri", last_modified=self._NOW)]}))
airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py:    def test_given_exception_when_read_records_from_slice_then_do_process_other_files(
airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py:            self._stream.read_records_from_slice(
airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py:    def test_given_traced_exception_when_read_records_from_slice_then_fail(
airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py:                self._stream.read_records_from_slice(
airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py:    def test_given_exception_after_skipping_records_when_read_records_from_slice_then_send_warning(
airbyte-cdk/python/unit_tests/sources/file_based/stream/test_default_file_based_stream.py:            self._stream.read_records_from_slice(
```

Reply:
Updated.